src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/DesktopOsHelper.cs

```diff
@@ -44,14 +44,15 @@ public static bool IsLinux()
 
         public static bool IsMac()
         {
-#if MAC
-            return true;
+#if __MOBILE__
+            return false;
 #elif NETFRAMEWORK
             return Environment.OSVersion.Platform == PlatformID.MacOSX;
-#elif !__MOBILE__ 
-            return RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
+#elif NET8_0_OR_GREATER
+            string OSDescription = RuntimeInformation.OSDescription;
+            return OSDescription.Contains("Darwin", StringComparison.OrdinalIgnoreCase);
```

Comment:
Hmm, so is `RuntimeInformation.IsOSPlatform(OSPlatform.OSX)` not working on net8?

Reply:
No... at least `RuntimeInformation.IsOSPlatform(OSPlatform.OSX)` does not work on my mac.